megatron/core/extensions/transformer_engine.py

```diff
@@ -523,6 +575,181 @@ def sharded_state_dict(self, prefix='', sharded_offsets=(), metadata=None):
         )
 
 
+class TERowParallelLinearOp(te_ops.Sequential):
```

Comment:
Why do we need `TERowParallelLinearOp` module?

Reply:
tl;dr: this module is no longer necessary due to a recent refact at TE side. I have removed it;
This module was designed to make RowParallelLinear a TE op. Previously there were two code paths to implement row parallel linear, one is legacy and the other is TE operation-based API. The latter is what I needed. However, recent refact of TE has unified the two code paths and there is no longer legacy layer, so my wrapper class here is not necessary. I have removed this.